tweets/templatetags/tweet_tags.py

```diff
@@ -9,3 +9,7 @@ def can_edit_tweet(user, tweet):
 @register.filter
 def can_delete_tweet(user, tweet):
   return user.id == tweet.user_id
+
+@register.filter
+def can_retweet(user, tweet):
+  return user.id != tweet.user_id and user.is_authenticated and tweet.original_tweet is None
```

Comment:
- Make gaurd conditions. 
- Be mindful when using associations 
`tweet.original_tweet` will fire additional query to fetch `original_tweet`. Use `tweet.original_tweet_id`

Reply:
okay. Should I do this in all places I've used or just here for now.